x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/utils/utils.ts

```diff
@@ -426,17 +427,18 @@ export const getRuleRangeTuples = ({
   interval: string;
   maxSignals: number;
   ruleExecutionLogger: IRuleExecutionLogForExecutors;
+  getTimeRange: GetTimeRangeFn;
 }) => {
-  const originalFrom = dateMath.parse(from, { forceNow: startedAt });
-  const originalTo = dateMath.parse(to, { forceNow: startedAt });
-  if (originalFrom == null || originalTo == null) {
-    throw new Error('Failed to parse date math of rule.from or rule.to');
-  }
+  const startedAtString = new Date(startedAt).toISOString();
+
+  const { dateStart, dateEnd } = getTimeRange(from, startedAtString);
```

Comment:
I think from our side if the framework needs visibility into how the query time range is calculated then it would be easier if `getTimeRange` were just a library function rather than a rule executor param and `startedAt` and `interval` are faked as needed. Right now we don't use `interval` directly, but `from` is actually the `interval` + `additional lookback` from the UI - I could see migrating our rules to fix that weirdness and use `interval` directly, have a separate `lookback` or "drift tolerance", rule param, and let the framework run rules and influence the time range simply by setting the `interval`, `startedAt`, and any other applicable params as needed.

Let's chat on zoom!